### PR TITLE
Features to bring PHP SDK to v1

### PR DESCRIPTION
**Pull request by kgston**
- Fixes https://github.com/gyro-n/gopay-php-sdk/issues/2
- Adds missing relavent webhook event types
- Adds checkout info route
- Adds transfer status changes
- Adds subscription create, patch and delete
- Added additional webhook events
- Enumifcation
- Extra validation checks prior to request
  - Only includes checks that are guaranteed not to update soon (E.g. checks for terminal statuses) or 
- CVV patching, addition of metadata and customerId and listing of transaction tokens
- Configuration changes (display_time_zone, max charge & min charge for cvv check)
- Changed from websocket polling to http long polling
- Adds CircleCI stuff

### CODE REVIEW

File: src/Gopay/Utility/Json/JsonSchema.php [hunk 1]
```diff
@@ -105,6 +105,11 @@ public static function getField($json, $required, array $paths)
             throw new NoSuchPathException(null);
         }
         $nextKey = $paths[0];
+
+        if (is_bool($json)) {
```
Comment (adamsar) [inline src/Gopay/Utility/Json/JsonSchema.php]: Debugging stuff?

File: src/Gopay/Utility/RequesterUtils.php [hunk 2]
```diff
@@ -53,13 +53,25 @@ public static function executePost(
...
-        array $payload = array()
+        $payload = array()
     ) {
         $response = $requestContext->getRequester()->patch(
             $requestContext->getFullURL(),
             $payload,
             self::getHeaders($requestContext)
         );
+        if (is_bool($response)) {
+            echo $requestContext->getFullURL();
```
Comment (adamsar) [inline src/Gopay/Utility/RequesterUtils.php]: Assume more debugging stuff

File: src/Gopay/Enums/Currency.php [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Gopay\Enums;
+
+final class Currency extends TypedEnum
+{
+    // phpcs:disable
+    public static function JPY() { return self::create('JPY'); }
```
[the remark this answers is not in the document]
Comment (kgston) [reply src/Gopay/Enums/Currency.php]: Actually I based this list off the `PhysicalCurrency` object as I thought that represented the currencies the API accepts. 

How about using this implementation?
[moneyphp](https://github.com/moneyphp/money) | [docs](http://moneyphp.org/en/latest/features/currencies.html)

[the remark this answers is not in the document]
Comment (adamsar) [reply src/Gopay/Enums/Currency.php]: Basically that list is just useful for testing/spec purposes. I would go with that implementation. We may need to revise with more currency types like (e-monies)

[the remark this answers is not in the document]
Comment (kgston) [reply src/Gopay/Enums/Currency.php]: Apparently, the latest version of this package requires PHP 5.6. Can we just bump this up 2 minor versions?

**Pull request merged**